src/vs/workbench/api/common/extHostTerminalService.ts

```diff
@@ -550,10 +534,6 @@ export abstract class BaseExtHostTerminalService implements IExtHostTerminalServ
 	private _getTerminalByIdEventually(id: number, retries: number = 5): Promise<ExtHostTerminal | undefined> {
 		if (!this._getTerminalPromises[id]) {
 			this._getTerminalPromises[id] = this._createGetTerminalPromise(id, retries);
-		} else {
-			this._getTerminalPromises[id].then(c => {
-				return this._createGetTerminalPromise(id, retries);
-			});
```

Comment:
I think the purpose of this code was to handle the case where the promise `this._getTerminalPromises[id]` resolves to `undefined`. It definitely doesn't work currently but that might be a case that's still an issue.

Reply:
@Tyriar But it doesn't return anything, so no effect. Or was there any side effect of invoking _getTerminalByIdEventually in code before some of the changes?

Reply:
Yeah it definitely didn't work so it's fine to remove, I think we're still missing that case though. Ideally the resolve function would be kept around, similar to how id promise works on BaseExtHostTerminal:

https://github.com/microsoft/vscode/blob/9e899b556f10f09217239396af8ee95009612982/src/vs/workbench/api/common/extHostTerminalService.ts#L41-L42